PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverActivity.java

```diff
@@ -114,6 +114,20 @@ public void onClick(View v) {
                 startActivity(new Intent(ScenarioOverActivity.this, GameActivity.class));
             }
         });
+
+        replayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                SessionHistory.currSessionID = SessionHistory.prevSessionID;
+                SessionHistory.totalPoints -= SessionHistory.currScenePoints;
+                SessionHistory.currScenePoints = 0;
+                scenarioActivityDone = 0;
+                DatabaseHandler dbHandler = new DatabaseHandler(ScenarioOverActivity.this);
+                dbHandler.resetCompleted(SessionHistory.currSessionID);
+                dbHandler.resetReplayed(SessionHistory.currSessionID);
+                startActivity(new Intent(ScenarioOverActivity.this, GameActivity.class));
```

Comment:
So a new activity is launched here without finishing the last one which leads to the same scene being created as many times as replay is clicked. Multiple games should not be present in the back-stack. Please fix.